Create new branch for V8 5.1

ofrobots opened the pull request:
<!--
Thank you for submitting a pull request to Node.js. Before you submit, please
review below requirements and walk through the checklist. You can 'tick'
a box by using the letter "x": [x].

Run the test suite by invoking: `make -j4 lint test` on linux or
`vcbuild test nosign` on Windows.

If this aims to fix a regression or you’re adding a feature, make sure you also
write a test. Finally – if possible – a benchmark that quantifies your changes.

Finally, read through our contributors guide and make adjustments as necessary:
https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist

<!-- remove lines that do not apply to you -->

- [x] tests and code linting passes
- [x] the commit message follows commit guidelines


##### Affected core subsystem(s)

<!-- provide affected core subsystem(s) (like doc, cluster, crypto, etc) -->
deps, test

##### Description of change

<!-- provide a description of the change below this comment -->
deps: upgrade to V8 5.1.281.24 [vee-eight-5.1]

Pick up the latest branch-head for V8 5.1. This branch brings in
improved language support and performance improvements. For full
details: http://v8project.blogspot.com/2016/04/v8-release-51.html

* Picks up the latest branch head for 5.1 [1]
* Edit v8 gitignore to allow trace_event copy
* Update V8 DEP trace_event as per deps/v8/DEPS [2]
* Fix tests.

[1] v8/v8@f81c34e
[2] https://chromium.googlesource.com/chromium/src/base/trace_event/common/+/c8c8665c2deaf1cc749d9f8e153256d4f67bf1b8

R=@nodejs/v8
CI: https://ci.nodejs.org/job/node-test-pull-request/2440/

ofrobots commented:
@misterdjules Do you have any ideas about the dtrace compile issue on SmartOS: https://ci.nodejs.org/job/node-test-commit-smartos/2266/nodes=smartos14-32/console

bnoordhuis commented:
@misterdjules `StandardFrameConstants::kMarkerOffset` was replaced with `CommonFrameConstants::kContextOrFrameTypeOffset`.  V8's tools/gen-postmortem-data.py and node's src/v8abbr.h and src/v8ustack.d need updating.

For reference: v8/v8@9dcd0857d64a117801bf91cb6caf7d6578809cb1 and v8/v8@cbd91c52088bf953ac529ba6d07084e8b818e15c.  The commit log mentions `StandardFrameConstants::kFrameOffset` but I don't think that was ever a thing.  Maybe a leftover from an earlier revision of the CL.

@ofrobots You might want to float this patch for now to get past the build error:
```diff
diff --git a/configure b/configure
index 03a889d..8839e12 100755
--- a/configure
+++ b/configure
@@ -736,7 +736,7 @@ def configure_node(o):
   if flavor in ('solaris', 'mac', 'linux', 'freebsd'):
     use_dtrace = not options.without_dtrace
     # Don't enable by default on linux and freebsd
-    if flavor in ('linux', 'freebsd'):
+    if flavor in ('linux', 'freebsd', 'solaris'):
       use_dtrace = options.with_dtrace
 
     if flavor == 'linux':
```
I've tried restarting the armv8 build job but the CI is very, very slow right now.

ofrobots commented:
Temporarily disabled dtrace by default on solaris, as per suggestion from @bnoordhuis. New CI: https://ci.nodejs.org/job/node-test-pull-request/2456/.

mhdawson commented:
For my own gratification, CI run on linuxOne: https://ci.nodejs.org/job/node-test-commit-linuxone-mdawson/nodes=rhel72-s390x/22/, all green :)

ofrobots commented:
It turns out that
* @davepacheco already provided a potential fix for this: https://github.com/nodejs/post-mortem/issues/21#issuecomment-200018977, and
* I already fixed gen-postmortem-data.py upstream when the V8 lkgr + node.js integration builds caught this originally.

I have updated the PR with the patch from @davepacheco, and removed the temporary fix. New CI: https://ci.nodejs.org/job/node-test-pull-request/2461/

ofrobots commented:
CI is green. @nodejs/v8 please review.

bnoordhuis commented:
Can you either use the full URL (`https://github.com/v8/v8/commit/879b617b`) or the v8/v8@879b617b syntax in second commit?  Otherwise LGTM.

ofrobots commented:
Thanks, fixed commit message, and landed on `vee-eight-5.1` as c2bbfe2...098d365.
